db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericStatement.java

```diff
@@ -620,6 +622,21 @@ else if (newMergeJoinString.equals("forced"))
                 }
                 cc.setCurrentTimestampPrecision(currentTimestampPrecision);
 
+                String timestampFormatString =
+                        PropertyUtil.getCachedDatabaseProperty(lcc, Property.SPLICE_TIMESTAMP_FORMAT);
+                if(timestampFormatString == null)
+                    cc.setTimestampFormat(CompilerContext.DEFAULT_TIMESTAMP_FORMAT);
+                else {
+                    try {
+                        // DB-10968 we shouldn't even allow setting this to the wrong value
+                        DateTimeFormatter.ofPattern(timestampFormatString);
+                    } catch(Exception e)
+                    {
+                        timestampFormatString = CompilerContext.DEFAULT_TIMESTAMP_FORMAT;
+                    }
+                    cc.setTimestampFormat(timestampFormatString);
+                }
+
                 String timestampPrecisionString =
```

Comment:
`timestampPrecisionString` can be removed along with the related logic from `GenericStatement`.